nixos/modules/services/x11/urserver.nix

```diff
@@ -0,0 +1,72 @@
+# urserver service
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.urserver;
+in {
+
+  options.services.urserver = {
+
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enable Unified Remote server.
+      '';
+    };
+
+    usedTCPPorts = mkOption {
```

Comment:
How do you tell the program what ports to use? As currently implemented this option doesn't add much value, the user can just set the `networking.firewall` option themselves.

Reply:
The server also spins up a webserver, the port configuration can be tweaked there.
It does mean that is something that the user has to tweak to taste anyway, so maybe the related nixos service options are indeed redundant. It does at least serve as a reminder that some ports need to be opened (something I ran into myself).

Reply:
If this is just a taste thing I would prefer to leave this in, as it makes sense to me that the ability to open the ports required for a service can be configured together with the service. Alternatively I would just open the ports to make sure it works out of the box anyway, but then you lose the flexibility to change the ports in the web interface and still have everything work. Also, for my situation I only use the WiFi option, so I can keep the other ports closed in my configuration.nix .

If this is a requirement for getting it accepted I will take it out of course.

Reply:
@SFrijters in my **opinion** when you can specify an `openFirewall` option combined with a `port`(s) option I think this is a great pattern. As this is It really just copies an existing option, providing no value on its own.

Maybe someone else will weigh in on this with their opinion too.

BTW there is a `types.port` which would be preferred if you're keeping this.

Reply:
Ok, I followed your suggestions. Didn't know about the `types.port` thing!

Reply:
@SFrijters See discussion at https://github.com/NixOS/rfcs/pull/42 for why some people, my self included, are against adding redundant options - I claim the less is the better.